README.md

```diff
@@ -2,22 +2,36 @@
+terraform-plugin-framework is a module for building Terraform providers built on [terraform-plugin-go](https://github.com/hashicorp/terraform-plugin-go). It aims to provide as much of the power, predictability, and versatility of terraform-plugin-go as it can while abstracting away implementation details and repetitive, verbose tasks.
+
+## Status
+
+terraform-plugin-framework is still experimental. We are committed to moving forward with the module, but cannot guarantee any of its interfaces will not change as long as it is in version 0. We're waiting for more feedback, usage, and maturity before we're comfortable committing to APIs with the same years-long support timelines that [terraform-plugin-sdk](https://github.com/hashicorp/terraform-plugin-sdk) brings.
+
+terraform-plugin-framework is also not at full feature parity with terraform-plugin-sdk yet. Notably, it doesn't offer support for validation, modifying plans (including marking resources as needing to be recreated), importing resources, or upgrading resource state. We plan to add these features soon.
+
+We believe terraform-plugin-framework is still a suitable and reliable module to build Terraform providers on, and encourage community members that can afford occasional breaking changes to build with it. terraform-plugin-framework will eventually become a new major version of terraform-plugin-sdk, at which point its interfaces will be stable, but we need real-world use and feedback before we can be comfortable making those commitments. When that happens, this repository will be archived.
 
 We recommend only using tagged releases of this module, and examining the CHANGELOG when upgrading to a new release. Breaking changes will only be made in minor versions; patch releases will always maintain backwards compatibility.
 
 We welcome and appreciate issues and PRs discussing both the design and implementation of this module.
 
 ## Terraform CLI Compatibility
 
-Plugins built with this framework are only compatible with Terraform versions above 0.12.0. Terraform 0.12.26 or higher is needed to run acceptance tests for providers built with this framework.
+Plugins built with this framework are only compatible with Terraform versions above v0.15.0. Terraform v0.15.4 or higher is needed to run acceptance tests for providers built with this framework.
```

Comment:
How about we simply pin at v0.15.4 and above?

Reply:
I just don't want to mislead people, because Terraform users 0.15.0 and higher will get the provider when `terraform init` is run, and I don't want someone to rely on it being only 0.15.4 users and above, if that makes sense.

Reply:
Could we also simplify compatibility promises and say that 1.0.0 and above is the tested CLI version since it provides its own compatibility promises, even if it is slightly off? Might be easier to centralize on that nice round major milestone than nitpick on 0.x versions.

Reply:
It's just that if we say it's 1.0+, we need to wait until June of 2022 for our official providers to adopt this. If we say 0.15.0, they can adopt it in April. (Plus or minus some wiggle room here and there because of major versions, etc etc)